### PR TITLE
add xitca-web

### PR DESCRIPTION
**Pull request by fakeshadow**
An experimental http library written in Rust.

### CODE REVIEW

File: frameworks/Rust/xitca-web/src/db.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        for _ in 1..=num {
+            let _ = write!(&mut q, "${},", pl);
+            pl += 1;
+        }
+        q.pop();
+        q.push(')');
+
+        let st = client.prepare(&q).await.unwrap();
+        updates.insert(num, st);
+    }
```
Comment (nbrady-techempower) [inline frameworks/Rust/xitca-web/src/db.rs:47]: Is this really creating 500 prepared update statements?

Comment (fakeshadow) [reply frameworks/Rust/xitca-web/src/db.rs:47]: Yes I believe it does prepare all possible update query and store them in client object. I copied this piece of code from actix bench and made slight change to make it compile.

Comment (nbrady-techempower) [reply frameworks/Rust/xitca-web/src/db.rs:47]: Ah, now I see that xitca is actix backwards...

File: frameworks/Rust/xitca-web/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+[package]
+name = "xitca-web"
+version = "0.1.0"
+edition = "2018"
+
+[dependencies]
+xitca-http = { git = "https://github.com/fakeshadow/xitca-web.git", branch = "tfb" }
```
Comment (nbrady-techempower) [inline frameworks/Rust/xitca-web/Cargo.toml:7]: Can you use the `rev` property here to lock all these down to specific commits? We want to make sure these aren't being updated without PRs here. Other than that, this looks ready to go.

Comment (fakeshadow) [reply frameworks/Rust/xitca-web/Cargo.toml]: Done.

**Pull request merged**